cli/command/inspect/inspector_test.go

```diff
@@ -219,3 +226,19 @@ func TestIndentedInspectorRawElements(t *testing.T) {
 		t.Fatalf("Expected `%s`, got `%s`", expected, b.String())
 	}
 }
+
+// moby/moby#32235
+func TestTemplateInspectorRawFallbackNumber(t *testing.T) {
```

Comment:
Does this test fail without the change?

I'm not sure I understand why the json decode decides to convert it to a float in some cases.

I'm also not sure this test actually tests the change.  Isn't `tryRawInspectFallback()` only called when inspect receives a type without those fields? It seems we're never creating a json object.

Could this test hit `tryRawInspectFallback()` directly instead. The smaller surface area makes it easier to tell what the test is exercising.

Maybe make this a table test and try it with a float as well, so that we can verify that a float is printed using the same literal?